tfsec 0.50.6 crashes in Azure DevOps pipeline

**Describe the bug**
tfsec crashes.
```log
panic: value is unknown

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.AsString(0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc30208, 0x110)
	/home/runner/work/tfsec/tfsec/vendor/github.com/zclconf/go-cty/cty/value_ops.go:1262 +0x165
github.com/aquasecurity/tfsec/internal/app/tfsec/block.(*HCLBlock).Clone(0xc00036e140, 0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc423f0, 0xc000b17ec0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/block/hclblock.go:65 +0x285
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).expandBlockCounts.func1(0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:216 +0x98
github.com/aquasecurity/tfsec/internal/app/tfsec/block.(*HCLAttribute).Each.func1(0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0x10662e0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/block/hclattribute.go:46 +0x7b
github.com/zclconf/go-cty/cty.Value.ForEachElement(0xc30320, 0xc00020a0f0, 0xae8da0, 0xc00093cdb0, 0xc00050cd38, 0xabb300)
	/home/runner/work/tfsec/tfsec/vendor/github.com/zclconf/go-cty/cty/value_ops.go:1149 +0x123
github.com/aquasecurity/tfsec/internal/app/tfsec/block.(*HCLAttribute).Each(0xc00021f850, 0xc0005c3560)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/block/hclattribute.go:45 +0x7f
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).expandBlockCounts(0xc0000b2120, 0xc0009ded00, 0xe, 0x10, 0xc0009ded00, 0xe, 0x10)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:215 +0x2f5
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).expandBlocks(0xc0000b2120, 0xc0005b8500, 0x10, 0x10, 0x0, 0x88, 0xae1f40)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:203 +0x79
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).EvaluateAll(0xc0000b2120, 0x13, 0xc00077e200, 0x34, 0xc0005b8500, 0x10)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:175 +0x32f
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).evaluateModules(0xc0000b2ab0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:127 +0x413
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).evaluateStep(0xc0000b2ab0, 0x0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:91 +0x6fe
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).EvaluateAll(0xc0000b2ab0, 0xb08906, 0x7, 0x13, 0xc000047400, 0x35)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:179 +0x4a5
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Parser).ParseDirectory(0xc000176f80, 0x13, 0xc00043fb90, 0x2, 0x2, 0xc000176f80)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/parser.go:106 +0xa5c
main.glob..func2(0x100a300, 0xc000081d10, 0x1, 0x5, 0x0, 0x0)
	/home/runner/work/tfsec/tfsec/cmd/tfsec/main.go:211 +0x69e
github.com/spf13/cobra.(*Command).execute(0x100a300, 0xc0000ac130, 0x5, 0x5, 0x100a300, 0xc0000ac130)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:856 +0x472
github.com/spf13/cobra.(*Command).ExecuteC(0x100a300, 0xc000042778, 0xc0002dff78, 0x406365)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:974 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:902
main.main()
	/home/runner/work/tfsec/tfsec/cmd/tfsec/main.go:89 +0x31
##[error]Bash exited with code '2'.
##[error]Bash wrote one or more lines to the standard error stream.
##[error]panic: value is unknown

goroutine 1 [running]:
github.com/zclconf/go-cty/cty.Value.AsString(0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc30208, 0x110)

##[error]	/home/runner/work/tfsec/tfsec/vendor/github.com/zclconf/go-cty/cty/value_ops.go:1262 +0x165
github.com/aquasecurity/tfsec/internal/app/tfsec/block.(*HCLBlock).Clone(0xc00036e140, 0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc423f0, 0xc000b17ec0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/block/hclblock.go:65 +0x285
github.com/aquasecurity/tfsec/internal/app/tfsec/parser.(*Evaluator).expandBlockCounts.func1(0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/parser/evaluator.go:216 +0x98
github.com/aquasecurity/tfsec/internal/app/tfsec/block.(*HCLAttribute).Each.func1(0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0xc30208, 0xc0000a60db, 0xa6ada0, 0x10662e0, 0x10662e0)
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/block/hclattribute.go:46 +0x7b
github.com/zclconf/go-cty/cty.Value.ForEachElement(0xc30320, 0xc00020a0f0, 0xae8da0, 0xc00093cdb0, 0xc00050cd38, 0xabb300)
```

**To Reproduce**
Steps to reproduce the behavior:
I'm not sure. Tfsec runs within an Azure DevOps pipeline and 0.50.2 was running fine, 0.50.6 crashes.

**Expected behavior**
tfsec doesn't crash.


**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04

--- a/internal/app/tfsec/block/hclblock.go
+++ b/internal/app/tfsec/block/hclblock.go
@@ -18,7 +18,7 @@ type HCLBlock struct {
 	evalContext *hcl.EvalContext
 	moduleBlock Block
 	expanded    bool
-	key         string
+	cloneIndex  int
 }
 
 func NewHCLBlock(hclBlock *hcl.Block, ctx *hcl.EvalContext, moduleBlock Block) Block {
@@ -57,15 +57,19 @@ func (block *HCLBlock) Clone(index cty.Value) Block {
 		for i := 0; i < len(labels); i++ {
 			labels[i] = clone.hclBlock.Labels[i]
 		}
-		switch index.Type() {
-		case cty.Number:
-			f, _ := index.AsBigFloat().Float64()
-			labels[position] = fmt.Sprintf("%s[%d]", clone.hclBlock.Labels[position], int(f))
-		case cty.String:
-			labels[position] = fmt.Sprintf("%s[%q]", clone.hclBlock.Labels[position], index.AsString())
-		default:
-			debug.Log("Invalid key type in iterable: %#v", index.Type())
-			labels[position] = fmt.Sprintf("%s[%#v]", clone.hclBlock.Labels[position], index)
+		if index.IsKnown() && !index.IsNull() {
+			switch index.Type() {
+			case cty.Number:
+				f, _ := index.AsBigFloat().Float64()
+				labels[position] = fmt.Sprintf("%s[%d]", clone.hclBlock.Labels[position], int(f))
+			case cty.String:
+				labels[position] = fmt.Sprintf("%s[%q]", clone.hclBlock.Labels[position], index.AsString())
+			default:
+				debug.Log("Invalid key type in iterable: %#v", index.Type())
+				labels[position] = fmt.Sprintf("%s[%#v]", clone.hclBlock.Labels[position], index)
+			}
+		} else {
+			labels[position] = fmt.Sprintf("%s[%d]", clone.hclBlock.Labels[position], block.cloneIndex)
 		}
 		clone.hclBlock.Labels = labels
 	}
@@ -74,6 +78,7 @@ func (block *HCLBlock) Clone(index cty.Value) Block {
 		"index": indexVal,
 	})
 	clone.markCountExpanded()
+	block.cloneIndex++
 	return clone
 }
 

